pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -403,7 +407,7 @@ public void downloadAndLoadSegment(SegmentZKMetadata zkMetadata, IndexLoadingCon
     String segmentName = zkMetadata.getSegmentName();
     _logger.info("Downloading and loading segment: {}", segmentName);
     File indexDir = downloadSegment(zkMetadata);
-    addSegment(ImmutableSegmentLoader.load(indexDir, indexLoadingConfig));
+    addSegment(ImmutableSegmentLoader.load(indexDir, indexLoadingConfig, _segmentPreprocessThrottler));
```

Comment:
does it make sense to add a TODO that the current `_segmentDownloadSemaphore` (the one to throttle segment download and segment decompression) should be made dynamic as well to avoid restarting servers when need to throttle them?

Reply:
done